### PR TITLE
Support host configurable max_cpu_frequency

### PR DESCRIPTION
**Pull request by yjian118**
This commit adds maxCPUFrequency in the hostprofile resource to support
the configuration of the the max_cpu_frequency on a host node.

Tests Passed:
1 Deploy a bare metal AIOSX w/o maxCPUFrequency configured. Generate
the deployment config with the deploy tool, the maxCPUFrequency is not
recorded.
2 Deploy a bare metal AIOSX with maxCPUFrequency configured. Validate
the max_cpu_frequncey value in the host post deploy. Generate the
deployment config with the deploy tool, the maxCPUFrequency is
recorded as configured.

Signed-off-by: Yuxing Jiang <yuxing.jiang@windriver.com>

### CODE REVIEW

File: config/crds/starlingx_v1_host.yaml [hunk 1]
```diff
@@ -670,6 +670,11 @@ spec:
                   description: Location defines the physical location of the host
                     in the data centre.
                   type: string
+                maxCPUFrequency:
+                  description: MaxCPUFrequency defines the maxium limit of the CPU
+                    frequency set on the host. + optional
+                  pattern: ^[1-9][0-9]*$
```
Comment (albailey) [inline config/crds/starlingx_v1_host.yaml:676]: As an FYI, this means you can never set this to zero.

File: config/crds/starlingx_v1_host.yaml [hunk 2]
```diff
@@ -670,6 +670,11 @@ spec:
                   description: Location defines the physical location of the host
                     in the data centre.
                   type: string
+                maxCPUFrequency:
+                  description: MaxCPUFrequency defines the maxium limit of the CPU
```
Comment (albailey) [inline config/crds/starlingx_v1_host.yaml:674]: typo.  maxium -> maximum

**Review by albailey — COMMENTED** [review]
minor typo and a small question about if we want to support 0

File: config/crds/starlingx_v1_host.yaml (hunk 1 above)
Comment (yjian118) [reply config/crds/starlingx_v1_host.yaml:676]: As per Iago, the max CPU frequency should always be above 0

File: config/crds/starlingx_v1_host.yaml (hunk 2 above)
Comment (yjian118) [reply config/crds/starlingx_v1_host.yaml:674]: Thanks, will update it.

**Review by albailey — APPROVED** [review]

**Pull request merged**